docs/guides/create-histogram.md

```diff
@@ -0,0 +1,192 @@
+description: Use the Nitric framework to easily build and deploy a serverless data visualization API for AWS, Google Cloud, or Azure
+---
+
+## What we'll be doing
+
+We'll be making a serverless application which can take information from a HTTP request and generate a histogram. This is a basic implementation which is completely open for extension. Optional extensions include being able to add a legend to the plot, or allowing new types of graphs such as a scatter plot or pie chart.
+
+## Prerequisites
+
+- [Pipenv](https://pypi.org/project/pipenv/) - for simplified dependency management
+- The [Nitric CLI](https://nitric.io/docs/installation)
+- An [AWS](https://aws.amazon.com), [GCP](https://cloud.google.com) or [Azure](https://azure.microsoft.com) account (_your choice_)
+
+## Getting started
+
+We’ll start by creating a new project for our API.
+
+```bash
+nitric new
+```
+
+Create a project, name it and select your preferred starter template.
+
+```bash
+? What is the name of the project? histogram-api
+? Choose a template: official/Python - Starter
+```
+
+Next, open the project in your editor of choice.
+
+```bash
+> cd histogram-api
+```
+
+Make sure all dependencies are resolved using Pipenv:
+
+```bash
+pipenv install --dev
+```
+
+Starting from scratch in the `hello.py` function file, lets start by importing and defining our api.
+
+```python
+from nitric.resources import api
+from nitric.application import Nitric
+
+mainApi = api("main")
+
+Nitric.run()
+```
+
+We can then define our api route which will accept the histogram data to then be returned as an image.
+
+```python
+from nitric.resources import api
+from nitric.application import Nitric
+
+mainApi = api("main")
+
+@mainApi.get("/histogram")
+async def create_histogram(ctx):
+  pass
+
+Nitric.run()
+```
+
+The route will be registered for a GET method so that it is a link that can be embed. For example, once the api is complete, it will be able to be used in situations like:
+
+```html
+<img
+  src="http://localhost:4001/histogram?data=1,1,2,2,2,3,3,3,3,5,6,4,5,3,2,6,6,6,5,4&ylabel=frequency&xlabel=dice rolls&title=Experimental probablity of 20 dice rolls"
+/>
+```
+
+We can then write the logic for creating the histogram from the request. As seen in the above url, there are 4 query parameters that we want a user to be able to configure, the data, y-label, x-label, and title.
+
+```python
+@mainApi.get("/histogram")
+async def create_histogram(ctx):
+  # Extract the comma-delimited data and split it into array
+  data = ctx.req.query.get("data")[0].split(',')
+
+  # Get the optional histogram labels
+  xlabel = ctx.req.query.get("xlabel")
+  ylabel = ctx.req.query.get("ylabel")
+  title = ctx.req.query.get("title")
+
+  try:
+    data = [int(d) for d in data]
+    data.sort()
+  except ValueError:
+    # If casting to int throws error, return a bad request status code
+    ctx.res.status = 400
+    return
+
+  plt.hist(x=data)
```

Comment:
plt is used here without being imported earlier. I'm assuming it's missing something like `import matplotlib.pyplot as plt` earlier in the example?